spec/services/onestop_id_spec.rb

```diff
@@ -1,64 +1,50 @@
     expect {
-      OnestopId.new('Retiro Station', '-58.374722 -34.591389')
+      TestOnestopId.new('Retiro Station', '-58.374722 -34.591389')
     }.to raise_error(ArgumentError)
 
     expect {
-      OnestopId.new('Retiro Station', [-58.374722,-34.591389])
+      TestOnestopId.new('Retiro Station', [-58.374722,-34.591389])
+    }.to raise_error(ArgumentError)
+    expect {
+      TestOnestopId.new(name: 'Retiro Station')
+    }.to raise_error(ArgumentError)
+    expect {
+      TestOnestopId.new(geohash: '9q9')
     }.to raise_error(ArgumentError)
   end
 
-  context 'create' do
+  context '#geohash' do
     it 'filters geohashes' do
       expect(
-        OnestopId.new(entity_prefix: 's', geohash: 'a9q9', name: 'test').to_s
+        TestOnestopId.new(geohash: 'a9q9', name: 'test').to_s
       ).to eq('s-9q9-test')
     end
-    it 'filters names' do
-      expect(
-        OnestopId.new(entity_prefix: 's', geohash: '9q9', name: 'Foo Bar!').to_s
-      ).to eq('s-9q9-foobar')
-    end
-    it 'filters names at' do
-      expect(
-        OnestopId.new(entity_prefix: 's', geohash: '9q9', name: 'foo@bar').to_s
-      ).to eq('s-9q9-foo~bar')
-    end
   end
 
-  context 'validation' do
-    it 'must not be blank' do
-      is_a_valid_onestop_id, errors = OnestopId.validate_onestop_id_string('')
-      expect(is_a_valid_onestop_id).to be false
-      expect(errors).to include 'must not be blank'
-    end
-    it 'must start with "s-" as its 1st component' do
-      is_a_valid_onestop_id, errors = OnestopId.validate_onestop_id_string('69y7pwu-RetSta', expected_entity_type: 'stop')
-      expect(is_a_valid_onestop_id).to be false
-      expect(errors).to include 'must start with "s" as its 1st component'
-    end
-
-    it 'must include 3 components separated by hyphens ("-")' do
-      is_a_valid_onestop_id, errors = OnestopId.validate_onestop_id_string('s-69y7pwuRetSta')
-      expect(is_a_valid_onestop_id).to be false
-      expect(errors).to include 'must include 3 components separated by hyphens ("-")'
+  context '#name' do
+    it 'allows only letters, digits, ~, @ in name' do
+      expect(TestOnestopId.new(geohash: '9q9', name: 'foo bar').to_s).to eq('s-9q9-foobar')
+      expect(TestOnestopId.new(geohash: '9q9', name: 'foo bar!').to_s).to eq('s-9q9-foobar')
+      expect(TestOnestopId.new(geohash: '9q9', name: 'foo~bar').to_s).to eq('s-9q9-foo~bar')
+      expect(TestOnestopId.new(geohash: '9q9', name: 'foo~bar0').to_s).to eq('s-9q9-foo~bar0')
     end
+  end
 
-    it 'must include a valid geohash as its 2nd component, after "s-"' do
-      is_a_valid_onestop_id, errors = OnestopId.validate_onestop_id_string('s-69y@7pwu-RetSta')
-      expect(is_a_valid_onestop_id).to be false
-      expect(errors).to include 'must include a valid geohash as its 2nd component'
+  context '#validate' do
+    it 'requires valid geohash' do
+      expect(TestOnestopId.new(geohash: 'a@', name: 'test').errors).to include 'invalid geohash'
```

Comment:
Line is too long. [96/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.